### PR TITLE
versioning

### LINKED ISSUE #112: spike a new python 3.11 docker image

### PR DESCRIPTION
**Pull request by bloodearnest**
- Add a new version of the python image.
- CI test and publish for versions

Fixes #112

### CODE REVIEW

File: scripts/publish.sh
Comment (evansd) [inline scripts/publish.sh:1]: This is a nifty bit of bash

File: scripts/.render.py.swp
Comment (evansd) [inline scripts/.render.py.swp:1]: More cruft

**Review by evansd — APPROVED** [review]
This is all very exciting. Sorry for taking so long to review. Beyond removing a couple of stray `.swp` files I don't have suggestions at this point.

File: tests/test_import.py [hunk 1]
```diff
@@ -1,21 +1,44 @@
...
+import re
 
 import pytest
 from pkg_resources import Requirement, get_provider
 
 
+# packages that have no way to detect their importable name
+BAD_PACKAGES = {
+    "beautifulsoup4": "bs4",
+    "protobuf": None, # AARRRRGG
```
[the remark this answers is not in the document]
Comment (bloodearnest) [reply tests/test_import.py:14]: At somepoint, they changed the importable name from `proto` to `protobuf` :shrug:

File: scripts/publish.sh
Comment (bloodearnest) [reply scripts/publish.sh:1]: lol, I don't know what this is referring to now!

Comment (evansd) [reply scripts/publish.sh:1]: It was the whole file!

**Pull request merged**